aiida/engine/processes/builder.py

```diff
@@ -48,14 +50,14 @@ def fgetter(self, name=name):
                     return self._data.get(name)
             elif port.has_default():
 
-                def fgetter(self, name=name, default=port.default):
+                def fgetter(self, name=name, default=port.default):  # pylint: disable=cell-var-from-loop
                     return self._data.get(name, default)
             else:
 
                 def fgetter(self, name=name):
                     return self._data.get(name, None)
 
-            def fsetter(self, value):
+            def fsetter(self, value, name=name):
```

Comment:
Would be good to add a comment to this method on why we are using `name=name`. Look at the amount of investigation we spent in the PR discussion to figure out why this is necessary. Just give a quick explanation that by binding the value as a default to a keyword argument, you are fixing the current value in the loop, because otherwise it would be scoped to the last value in the iteration.